plugins/eos/gs-plugin-eos.c

```diff
@@ -1772,6 +1772,40 @@ gs_plugin_eos_refine_popular_app (GsPlugin *plugin,
 	                            request_data);
 }
 
+static void
+gs_plugin_refine_proxy_app (GsPlugin	*plugin,
+			    GsApp	*app)
+{
+	g_autoptr(GsAppList) related_filtered = gs_app_list_new ();
+	GPtrArray *proxied_apps = gs_app_get_related (app);
+
+	for (guint i = 0; i < proxied_apps->len; ++i) {
+		GsApp *proxied_app = g_ptr_array_index (proxied_apps, i);
+
+		if (gs_app_get_state (proxied_app) == AS_APP_STATE_AVAILABLE ||
+		    gs_app_is_updatable (proxied_app)) {
+			g_debug ("App %s has an update or needs to be installed/updated "
+				 "by the proxy app %s",
+				 gs_app_get_unique_id (proxied_app),
+				 gs_app_get_unique_id (app));
+			gs_app_list_add (related_filtered, proxied_app);
+		}
+	}
+
+	gs_app_clear_related (app);
+
+	for (guint i = 0; i < gs_app_list_length (related_filtered); ++i) {
+		GsApp *related_app = gs_app_list_index (related_filtered, i);
+		gs_app_add_related (app, related_app);
+	}
+
+	gs_app_set_state (app, AS_APP_STATE_UNKNOWN);
```

Comment:
If `proxied_apps->len == 0`, then the function will mark the app as `STATE_UNKNOWN` unnecessarily (and will otherwise be a no-op). Is that correct?

Reply:
This is fine in any case and deliberate, as an unknown state ensures the proxy app will never be shown anywhere.

Reply:
Maybe that could be added in a comment above the `gs_app_set_state()` call.

Reply:
I will add that now.